dev/com.ibm.ws.messaging.comms.client/src/com/ibm/ws/sib/jfapchannel/richclient/framework/impl/RichClientTransportFactory.java

```diff
@@ -79,31 +85,44 @@ public NetworkConnectionFactory getOutboundNetworkConnectionFactoryByName(String
 
         try
         {
-            // Get the virtual connection factory from the channel framework using the chain name
-
-            VirtualConnectionFactory vcFactory = CommsClientServiceFacade.getChannelFramewrok().getOutboundVCFactory(chainName);
+        	// Get the virtual connection factory from the channel framework using the chain name
+        	if(CommsOutboundChain.getChainDetails(chainName) != null) {
+        		if(CommsOutboundChain.getChainDetails(chainName).useNetty()) {
+        			// If Netty, create new Netty channel factory
+        			// TODO: Consult with team. Getting error difference cause channelfw fails for testSSLFeatureUpdate
+        			// in com.ibm.ws.messaging.open_comms_fat because SSL chain failed to init properly due to no SSL Options
+        			// Check what to do here appropriately
+        			CommsOutboundChain chain = CommsOutboundChain.getChainDetails(chainName);
+        			boolean usingSSL = chain.isSSL();
+        			if(usingSSL && chain.getSslOptions() == null)
```

Comment:
I decided to throw an InvalidChainNameException `Chain configuration not found in framework` since that is what is thrown with Channelfw to match it accordingly since some tests were failing without the exception. Unlike Channelfw where it fails to add the chain if SSLOptions are not present and this is the way it's figured out I like that with Netty we can tell this is the cause so `Chain is set to use SSL but SSL config options not present` seems better. I'll check if I can make this switch

Reply:
I am happy to leave the decision to you.